pentagon/cli.py

```diff
@@ -65,18 +74,111 @@ def start_project(name, **kwargs):
+
+
+def _run(action, component_path, additional_args, options):
+    logging.basicConfig(level=options.get('log_level'))
+    logging.debug("Importing module Pentagon{}".format(component_path))
+    logging.debug("with options: {}".format(options))
+    logging.debug("and additional arguments: {}".format(additional_args))
+
+    component_class = get_component_class(component_path)
+
+    # Process infile or data options
+    data = {}
     try:
-        pentagon.PentagonComponent(name, kwargs).install()
+        file = options.get('file', None)
+        if file is not None:
+            data = parse_infile(file)
+        data = parse_data(options.get('data', data))
+
     except Exception as e:
+        logging.error("Error parsing data from file or -D arguments")
+        logging.error(e)
+        logging.debug(traceback.print_exc(e))
+
+    try:
+        if action == 'get':
+            component_class(data, additional_args).get(options.get('out'))
+        if action == 'add':
+            component_class(data, additional_args).add(options.get('out'))
+    except Exception, e:
         logging.error(e)
         logging.debug(traceback.print_exc(e))
 
+
 # Making names more terminal friendly
-cli.add_command(install_component, "install-component")
 cli.add_command(start_project, "start-project")
+cli.add_command(add, "add")
+cli.add_command(get, "get")
+
+
+def get_component_class(component_path):   # Construct Class path from component input
+    component_path_list = component_path.split(".")
+    if len(component_path_list) > 1:
+        component_name = ".".join(component_path.split(".")[0:-1])
+        component_class_name = component_path.split(".")[-1].title()
+    else:
+        component_name = component_path
+        component_class_name = component_path.title()
+
+    # Find Class if it exists
+    component_class = locate("pentagon.component.{}.{}".format(component_name, component_class_name))
+    if component_class is None:
+        component_class = locate("pentagon{}.{}".format(component_name, component_class_name))
+
+    if component_class is None:
+        logging.error("No module or class found: {}".format(component_path))
+
+    return component_class
+
+
+def parse_infile(file):
+    """ Parse data structure from file into dictionary for component use """
+    print file
+    with open(file) as data_file:
+        # try:
+        #     data = json.load(data_file)
+        #     return data
+        # except ValueError as json_error:
+        #     pass
+
+        try:
+            data = yaml.load(data_file)
+            print data
+            return data
+        except yaml.YAMLError as yaml_error:
+            pass
+
+    logging.error("Unable to parse in file. {}{} ".format(json_errror, yaml_error))
+
+
+def parse_data(data, d={}):
+    """ Function to parse the incoming -D options into a dict """
+    for kv in data:
+        key = kv.split('=')[0]
+        val = kv.split('=')[1]
```

Comment:
Is an `IndexError` raised here if the data is missing any `=`? If so can error checking be added to make it a bit prettier to the user? 

This also looks like it might mangle data that explicitly has a `=` in the value. 

```
>>> 'a=b=c'.split('=')
['a', 'b', 'c']
```

Using 

```
key, value = kv.split('=', 1)
```

will preserve the `=` in the value.

Reply:
I want to leave room for 'key only' -D options but better '=' is added